src/thing_creature.c

```diff
@@ -1815,11 +1815,16 @@ TngUpdateRet process_creature_state(struct Thing *thing)
                 struct Thing* doortng = get_door_for_position(x, y);
                 if (!thing_is_invalid(doortng))
                 {
-                    if (thing->owner != doortng->owner) {
+                    if (thing->owner != doortng->owner)
+                    {
                         set_creature_door_combat(thing, doortng);
+                        cctrl->collided_door_subtile = 0;
                     }
                 }
-                cctrl->collided_door_subtile = 0;
+                else
+                {
+                    cctrl->collided_door_subtile = 0;
+                }
```

Comment:
I think it would be a bit easier to read if written in a way that only uses the cctrl->collided_door_subtile = 0; line once

Reply:
I can't think of a way to have it be more clear. I added a condition (functional change) and a comment. Let me know if it's now easy enough to read.

Reply:
received confirmation of that on discord, merging.